R/utils.R

```diff
@@ -6,7 +6,14 @@
 #'
 #' @examples
 get_margins <- function (chars) {
-  return (trimws(strsplit(chars, ":")[[1]]))
+  chars <- trimws(chars)
+  left_is_exclusive <- substr(chars, 1, 1) == "("
+  right_is_exclusive <- substr(chars, nchar(chars), nchar(chars) + 1) == ")"
+
+  margins <- strsplit(substr(chars, 2, nchar(chars) - 1), ",")[[1]]
+  if (left_is_exclusive) margins[1] = as.character(as.numeric(margins[1]) + 1);
```

Comment:
Remember an open interval does not imply that the margins needs to be reduced by 1. For e.g., a recFrom value of (1,2] means that the effective range us 0.99999....,2 whereas in your case it would be 0 to 2 which is wrong.
Same issue with the next line of code.